prowler/providers/gcp/gcp_provider.py

```diff
@@ -316,9 +324,16 @@ def test_connection(
                     client_id, client_secret, refresh_token
                 )
 
-            session, _ = GcpProvider.setup_session(
+            session, project_id = GcpProvider.setup_session(
                 credentials_file, service_account, gcp_credentials
             )
+
+            if provider_id and project_id != provider_id:
+                # Logic to check if the provider ID matches the project ID
+                GcpProvider.validate_project_id(
```

Comment:
I think it'll be easier if you include the code within the function, so it does not match raise Exception and if matches return `connnected = True`. What do you think?

Reply:
I would prefer to modularize the code into the `validate_project_id` method, this way I can document what the method does in order to clarify the usage for the user.